platforms/documentation/docs/src/docs/userguide/dep-man/04-modeling-features/artifact_transforms.adoc

```diff
@@ -212,11 +361,37 @@ include::sample[dir="snippets/dependencyManagement/artifactTransforms-relocate/g
 <5> Inject the input artifact dependencies
 <6> Use the parameters
 
-== Registering artifact transforms
+Note the classes to be relocated are determined by examining the packages of the input artifact and its dependencies.
+Additionally, the transform ensures that packages contained in JAR files on an external classpath are not relocated.
+
+== Implementing incremental Artifact Transforms
+
+Similar to <<custom_tasks#incremental_tasks,incremental tasks>>, Artifact Transforms can avoid work by only processing changed files from the last execution.
+This is done by using the link:{groovyDslPath}/org.gradle.work.InputChanges.html[InputChanges] interface.
+
+For Artifact Transforms, only the input artifact is an incremental input; therefore, the transform can only query for changes there.
+To use link:{groovyDslPath}/org.gradle.work.InputChanges.html[InputChanges] in the transform action, inject it into the action.
 
-You need to register the artifact transform actions, providing parameters if necessary, so that they can be selected when resolving dependencies.
+For more information on how to use link:{groovyDslPath}/org.gradle.work.InputChanges.html[InputChanges], see the corresponding documentation for <<custom_tasks#incremental_tasks,incremental tasks>>.
 
-In order to register an artifact transform, you must use link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyHandler.html#org.gradle.api.artifacts.dsl.DependencyHandler:registerTransform(java.lang.Class,%20org.gradle.api.Action)[registerTransform()] within the `dependencies {}` block.
+Here is an example of an incremental transform that counts the lines of code in Java source files:
+
+====
+include::sample[dir="snippets/dependencyManagement/artifactTransforms-incremental/kotlin",files="build.gradle.kts[tags=artifact-transform-countloc]"]
+include::sample[dir="snippets/dependencyManagement/artifactTransforms-incremental/groovy",files="build.gradle[tags=artifact-transform-countloc]"]
+====
+<1> Inject `InputChanges`
+<2> Query for changes in the input artifact
+
+// TODO: https://github.com/gradle/gradle/issues/10867
+// [[artifact-resolution-queries]]
+// == Artifact resolution queries
+
+== Registering Artifact Transforms
```

Comment:
After going through this whole document, I am surprised the registering section is at the end. 
I think moving this section to before we describe _how_ to implement a transform will give the user a much better idea of how they're used.

Reply:
I am keeping the same order as the overview for continuity and to make it easy to read.